jerry-core/ecma/base/ecma-globals.h

```diff
@@ -55,8 +55,8 @@
  */
 typedef enum
 {
-  ECMA_TYPE_SIMPLE, /**< simple value */
-  ECMA_TYPE_NUMBER, /**< 64-bit integer */
+  ECMA_TYPE_NON_PTR_VALUE, /**< directly encoded value, a 28 bit signed integer or a simple value */
+  ECMA_TYPE_FLOAT_NUMBER, /**< pointer to a 64 or 32 bit floating point number */
```

Comment:
Ok. I used this name first, but I didn't particularly like it. But I am not against it either.